src/transactions.rs

```diff
@@ -454,16 +442,17 @@ impl<L: Ledger> Transactions<L> {
     #[allow(clippy::too_many_arguments)]
     pub fn create(
         &mut self,
+        uid: TransactionUID<L>,
+        hash: TransactionHash<L>,
         params: TransactionParams<L>,
     ) -> Result<TransactionEditor<'_, L>, KeystoreError<L>> {
         let txn = Transaction::<L> {
-            uid: params.uid.unwrap(),
+            uid,
             timeout: params.timeout,
-            hash: None,
+            hash: Some(hash),
```

Comment:
Does `hash` still need to be an `Option`, if we get it as a parameter for `create`? (BTW when we do #238, `hash` and `uid` will always be the same, so we can remove one of them and this will get a bit simpler)

Reply:
The only reason I left the hash in was because I thought for some other ledger types Hash and UID could be different.  That said now that I am looking at it is seems like there isn't any way to decouple them.  I think removing UID makes sense since it is derived from the hash.

Reply:
Decided to keep UID in the transaction because it has some stuff already built in and will be easier to extend if we need to because it's defined in this repo.